config/config.go

```diff
@@ -474,6 +483,34 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// MarathonSDConfig is the configuration for services running on Marathon
+type MarathonSDConfig struct {
+	Name           string   `yaml:"name,omitempty"`
```

Comment:
This seems to be used to prefix the job label and such but not be necessary to interact with the Marathon API.
If so, I assume this is a requirement for your setup and it should rather be solved via relabeling than be integrated into a SD mechanism.